src/builtins/core/date.rs

```diff
@@ -600,8 +600,52 @@ impl PlainDate {
             .with_calendar(self.calendar.identifier(), display_calendar)
             .build()
     }
+
+    #[inline]
+    pub fn to_zoned_date_time(
+        &self, 
+        tz: TimeZone,
+        pt: Option<PlainTime>, 
+        provider: &impl TimeZoneProvider
+    ) -> TemporalResult<ZonedDateTime> {
+        //3. If item is an Object, then
+            //a. Let timeZoneLike be ? Get(item, "timeZone").
+            //b. If timeZoneLike is undefined, then
+            //   i. Let timeZone be ? ToTemporalTimeZoneIdentifier(item).
+            //    ii. Let temporalTime be undefined.
+            // c. Else,
+            //    i. Let timeZone be ? ToTemporalTimeZoneIdentifier(timeZoneLike).
+            //    ii. Let temporalTime be ? Get(item, "plainTime").
+        // 4. Else,
+        //     a. Let timeZone be ? ToTemporalTimeZoneIdentifier(item).
+        //     b. Let temporalTime be undefined.
+
+        //  5. If temporalTime is undefined, then
+        //     a. Let epochNs be ? GetStartOfDay(timeZone, temporalDate.[[ISODate]]).
+
+        //  6. Else,
+        //     a. Set temporalTime to ? ToTemporalTime(temporalTime).
+        //     b. Let isoDateTime be CombineISODateAndTimeRecord(temporalDate.[[ISODate]], temporalTime.[[Time]]).
+        //     c. If ISODateTimeWithinLimits(isoDateTime) is false, throw a RangeError exception.
+        //     d. Let epochNs be ? GetEpochNanosecondsFor(timeZone, isoDateTime, compatible).
+        //  7. Return ! CreateTemporalZonedDateTime(epochNs, timeZone, temporalDate.[[Calendar]]).
+
+        let result_iso = if let Some(time) = pt {
```

Comment:
issue: this is close the the general structure of the Steps 5 and 6, but the return should be `epoch_ns`.

The steps above should be able to be followed nearly as is. `TimeZone::get_start_of_day` can be called in 5.a. (the plain_time is None case). 6.a. and 6.b. can be handled by `IsoDateTime::new`.

Take a stab at it and feel free to reach out on matrix if you have any questions.